backend/app/api/v1/commons/ocp.py

```diff
@@ -2,51 +2,196 @@
+        await es.close()
+        tasks = [item['_source'] for item in response["data"]]
+        jobs = pd.json_normalize(tasks)
+        if len(jobs) == 0:
+            return jobs
+
+        jobs[['masterNodesCount', 'workerNodesCount',
+            'infraNodesCount', 'totalNodesCount']] = jobs[['masterNodesCount', 'workerNodesCount', 'infraNodesCount', 'totalNodesCount']].fillna(0)
+        jobs.fillna('', inplace=True)
+        jobs[['ipsec', 'fips', 'encrypted',
+            'publish', 'computeArch', 'controlPlaneArch']] = jobs[['ipsec', 'fips', 'encrypted',
+                                                                    'publish', 'computeArch', 'controlPlaneArch']].replace(r'^\s*$', "N/A", regex=True)
+        jobs['encryptionType'] = jobs.apply(fillEncryptionType, axis=1)
+        jobs['benchmark'] = jobs.apply(utils.updateBenchmark, axis=1)
+        jobs['platform'] = jobs.apply(utils.clasifyAWSJobs, axis=1)
+        jobs['jobType'] = jobs.apply(utils.jobType, axis=1)
+        jobs['isRehearse'] = jobs.apply(utils.isRehearse, axis=1)
+        jobs['jobStatus'] = jobs.apply(utils.updateStatus, axis=1)
+        jobs['build'] = jobs.apply(utils.getBuild, axis=1)
+
+        cleanJobs = jobs[jobs['platform'] != ""]
+
+        jbs = cleanJobs
+        jbs['shortVersion'] = jbs['ocpVersion'].str.slice(0, 4)
+
+        return ({"data":jobs,"total":response["total"]})
+    
+    except Exception as err:
+        print(f"{type(err).__name__} was raised14: {err}") 
+
+def getMatchPhrase(key, item):
+    buildObj = {"match_phrase": {key: item}}  
+    return buildObj
+    
+def getSummary(jobStatus, isFilterReset):
+    if isFilterReset:
+        new_dict = {item['key']:0 for item in jobStatus}
+    else:
+        new_dict = {item['key']:item['doc_count'] for item in jobStatus}     
+    return new_dict
+
+async def getFilterData(start_datetime: date, end_datetime: date, size:int, offset:int, sort:str, filter:str, configpath: str):
+    try:
+        start_date = start_datetime.strftime('%Y-%m-%d') if start_datetime else (datetime.utcnow().date() - timedelta(days=5).strftime('%Y-%m-%d'))
+        end_date = end_datetime.strftime('%Y-%m-%d') if end_datetime else datetime.utcnow().strftime('%Y-%m-%d')
+        query = {"aggs":{"min_timestamp":{"min":{"field":start_date}},"max_timestamp":{"max":{"field":end_date}}},"query":{"bool":{"filter":[{"range":{"timestamp":{"format":"yyyy-MM-dd","lte":end_date,"gte":start_date}}}],"should":[],"must_not":[]}}}
+        
+        es = ElasticService(configpath=configpath)
+        if bool(sort):
+            query.update({"sort": json.loads(sort)})
+       
+        if bool(filter):
+            query = buildFilterQuery(filter, query) 
+            
+        aggregate = buildAggregateQuery()
+        query["aggs"].update(aggregate)
+        
+        isFilterReset = False
+        response = await es.filterPost(query=query)
```

Comment:
@MVarshini Did you make sure that this query is working?
```
RequestError was raised19: RequestError(400, 'search_phase_execution_exception', 'No mapping found for [build.keyword] in order to sort on')
TypeError was raised14: 'NoneType' object is not subscriptable
TypeError was raised15: argument of type 'NoneType' is not iterable
INFO:     127.0.0.1:42960 - "GET /api/v1/ocp/jobs?pretty=true&start_date=2024-10-10&end_date=2024-10-15&size=75&offset=1&sort=%7B%22build.keyword%22:%7B%22order%22:%22asc%22%7D%7D&filter=%7B%22ciSystem%22:[%22prow%22]%7D HTTP/1.1" 200 OK
Error retrieving filter data': RequestError(400, 'search_phase_execution_exception', 'No mapping found for [build.keyword] in order to sort on')
TypeError was raised: 'NoneType' object is not subscriptable
TypeError was raised: 'NoneType' object is not subscriptable
INFO:     127.0.0.1:42966 - "GET /api/v1/ocp/filters?pretty=true&start_date=2024-10-10&end_date=2024-10-15&size=75&offset=1&sort=%7B%22build.keyword%22:%7B%22order%22:%22asc%22%7D%7D&filter=%7B%22ciSystem%22:[%22prow%22]%7D HTTP/1.1" 200 OK
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7feb9ce77be0>, 441476.621)]']
connector: <aiohttp.connector.TCPConnector object at 0x7feb8308b430>
{'query': {'query_string': {'query': 'uuid: "bc78c231-6d7f-4726-8afb-ef4a87d84331"'}}}
{'ciSystem': 'PROW', 'uuid': 'bc78c231-6d7f-4726-8afb-ef4a87d84331', 'releaseStream': '4.15.0-0.nightly', 'platform': 'AWS', 'clusterType': 'self-managed', 'benchmark': 'cluster-density-v2', 'masterNodesCount': 3, 'workerNodesCount': 6, 'infraNodesCount': 3, 'masterNodesType': 'm6a.xlarge', 'workerNodesType': 'm6a.xlarge', 'infraNodesType': 'r5.xlarge', 'totalNodesCount': 12, 'clusterName': 'ci-op-yfrkcx95-96149-rzgmn', 'ocpVersion': '4.15.0-0.nightly-2024-10-11-141906', 'networkType': 'OVNKubernetes', 'buildTag': '1844750966205714432', 'jobStatus': 'success', 'buildUrl': 'https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-payload-control-plane-6nodes/1844750966205714432', 'upstreamJob': 'periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.15-nightly-x86-payload-control-plane-6nodes', 'upstreamJobBuild': '4.15.0-0.nightly-2024-10-11-141906-qe-perfscale', 'executionDate': '2024-10-11T16:08:02Z', 'jobDuration': '1634', 'startDate': '2024-10-11T16:08:02Z', 'endDate': '2024-10-11T16:35:16Z', 'startDateUnixTimestamp': '1728662882', 'endDateUnixTimestamp': '1728664516', 'timestamp': '2024-10-11T16:35:29Z', 'ipsec': 'false', 'ipsecMode': 'Disabled', 'fips': 'false', 'encrypted': 'false', 'encryptionType': '', 'publish': 'External', 'computeArch': 'amd64', 'controlPlaneArch': 'amd64'}
{'query': {'bool': {'must': [{'query_string': {'query': 'benchmark: "cluster-density-v2$" AND workerNodesType: "m6a.xlarge" AND masterNodesType: "m6a.xlarge" AND masterNodesCount: "3" AND workerNodesCount: "6" AND platform: "AWS" AND ocpVersion: 4.15* AND jobStatus: success'}}]}}}
{'query': {'query_string': {'query': '( uuid: "bc78c231-6d7f-4726-8afb-ef4a87d84331" ) AND metricName: "jobSummary"'}}}
{'query': {'query_string': {'query': '( uuid: "f1055fa3-8c5c-4cc9-b75d-39594c499a58" OR uuid: "4d59851e-b665-4c6e-914d-fa7fea1e2cb0" OR uuid: "ce07640a-a094-411a-84d2-f24d5cd6d064" OR uuid: "14a0d6ae-d4e4-435e-95ab-2a37b5633d3c" OR uuid: "c2c59959-7980-4d04-8495-c42d75957e28" OR uuid: "7335a811-95a3-453b-ab8d-182b6c1d1c3e" OR uuid: "44ee7997-9327-4dff-8d42-e64a40110cfe" OR uuid: "28e7d49f-535e-43ca-a383-feff6e669908" OR uuid: "d80cfd21-d005-4085-8f4f-57117a9a016f" OR uuid: "d3b26ef5-54a5-44df-b025-e6465e00a596" ) AND metricName: "jobSummary"'}}}
INFO:     127.0.0.1:51932 - "GET /api/v1/ocp/graph/bc78c231-6d7f-4726-8afb-ef4a87d84331 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/anyio/streams/memory.py", line 98, in receive
    return self.receive_nowait()
  File "/usr/local/lib/python3.9/site-packages/anyio/streams/memory.py", line 93, in receive_nowait
    raise WouldBlock
anyio.WouldBlock

```
I see a bunch of errors in the dashboard backend pod. Also it doesn't render frontend UI at all while applying filters.

Reply:
Just minimize the search this is only happening to me when looking at `cluster-density-v2` jobs in ocp tab, seems to be realted to the Graph of this type of jobs